### ISSUE TITLE
[R-Forge #2010] Define Ops.data.table for keyed data.tables

### ISSUE THREAD

**Issue by arunsrinivasan** [Rdatatable/data.table#639]
Submitted by: Matt Dowle; Assigned to: Nobody; [R-Forge link](https://r-forge.r-project.org/tracker/index.php?func=detail&aid=2010&group_id=240&atid=978)

Idea from this question:

http://stackoverflow.com/questions/10618837/joining-two-partial-data-tables-keeping-all-x-and-all-y

We do have Ops.data.table already, but it currently just dispatches to Ops.data.frame with a class change afterwards if cedta().

So, + (and other ops) should add together the intersecting key :

> DT1 = data.table(a=c(1,3,4,5), total=c(2,1,3,1), key="a")
> DT2 = data.table(a=c(2,3,5),total = c(5,1,2), key="a")
> DT1+DT2
> Error in Ops.data.frame(DT1, DT2) : 
> - only defined for equally-sized data frames

Enter a frame number, or 0 to exit   

1: Ops.data.table(DT1, DT2)
2: NextMethod()
3: Ops.data.frame(DT1, DT2)

Selection: 0

**Comment by jangorecki** [comment]
I don't think we really need such feature, the SO question also haven't got much attention in last 8 years. Voting to close.
